File: Task3-WebClient/eco-me-chan-front-end/src/services/IoTDeviceService.js
// src/services/IoTDeviceService.js
import apiInstance from './ApiService';

export const getUserIoTDevices = async (userId) => {
    try {
        const response = await apiInstance.get(`/api/IoTDevice/user/${userId}`);
        return response.data;
    } catch (error) {
        console.error('Error fetching user IoT devices:', error);
        throw error;
    }
};

export const getAllIoTDevices = async () => {
    try {
        const response = await apiInstance.get('/api/IoTDevice');
        return response.data;
    } catch (error) {
        console.error('Error fetching IoT devices:', error);
        throw error;
    }
};

export const deleteIoTDevice = async (deviceId) => {
    try {
        await apiInstance.delete(`/api/IoTDevice/${deviceId}`);
    } catch (error) {
        console.error('Error deleting IoT device:', error);
        throw error;
    }
};

export const updateIoTDevice = async (deviceId, deviceData) => {
    try {
        const response = await apiInstance.put(`/api/IoTDevice/${deviceId}`, deviceData);
        return response.data;
    } catch (error) {
        console.error('Error updating IoT device:', error);
        throw error;
    }
};

File: apzkr-pzpi-21-2-chan-vadym/Task4-MobileClient/EcoMeChan_Mobile/Services/IoTDeviceService.cs
// Services/IoTDeviceService.cs
using System.Collections.Generic;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;
using EcoMeChan_Mobile.Models;
using Newtonsoft.Json;

namespace EcoMeChan_Mobile.Services
{
    public class IoTDeviceService
    {
        private readonly HttpClient _httpClient;
        private readonly string _apiBaseUrl;

        public IoTDeviceService()
        {
            _httpClient = new HttpClient();
            _apiBaseUrl = "https://525f-149-88-110-59.ngrok-free.app"; 
        }

        public async Task<List<IoTDevice>> GetUserIoTDevices(int userId)
        {
            var response = await _httpClient.GetAsync($"{_apiBaseUrl}/api/IoTDevice/user/{userId}");
            var responseContent = await response.Content.ReadAsStringAsync();

            if (response.IsSuccessStatusCode)
            {
                var devices = JsonConvert.DeserializeObject<List<IoTDevice>>(responseContent);
                return devices;
            }
            else
            {
                // Обробка помилки отримання даних
                return null;
            }
        }

        public async Task<List<IoTDevice>> GetAllIoTDevicesAsync()
        {
            var response = await _httpClient.GetAsync($"{_apiBaseUrl}/api/IoTDevice");
            var responseContent = await response.Content.ReadAsStringAsync();

            if (response.IsSuccessStatusCode)
            {
                var devices = JsonConvert.DeserializeObject<List<IoTDevice>>(responseContent);
                return devices;
            }
            else
            {
                // Обробка помилки
                return null;
            }
        }

        public async Task<bool> UpdateIoTDeviceAsync(int deviceId, IoTDevice deviceData)
        {
            var json = JsonConvert.SerializeObject(deviceData);
            var content = new StringContent(json, Encoding.UTF8, "application/json");

            var response = await _httpClient.PutAsync($"{_apiBaseUrl}/api/IoTDevice/{deviceId}", content);

            return response.IsSuccessStatusCode;
        }

        public async Task<bool> DeleteIoTDeviceAsync(int deviceId)
        {
            var response = await _httpClient.DeleteAsync($"{_apiBaseUrl}/api/IoTDevice/{deviceId}");

            return response.IsSuccessStatusCode;
        }
    }
}